docs/contributor/arch/018-switch-from-gateways-to-a-central-agent.md

```diff
@@ -0,0 +1,72 @@
+  The gateway should provide autoscaling capabilities which are tricky to realize. It must be detected very fast if a scale out is needed and it situations were the backend is overloaded must be excluded from scale out.
+- every instance caches whole cluster state
+  Every instance of a gateway must be able to enrich data of every pod and with that need to cache all pod metadata
+- Istio mandatory for load balancing
+  Pushing OTLP data from an application to the gateway running with replicas requires loadbalancing. Unfortunately, a kubernetes service is acting on L4 and while GRPC/HTTP2 is on L/, so an additional loadbalancer is needed. That can be easily solved by using Istio, but it becomes a mandatory requirement for applications. However, there can be applications not using Istio (Statefulets often don't do) and like to provide OTLP data.
+- Istio is mandatory for mTLS
+  App to Gateway communication is across nodes and with that should use mTLS, so Istio needs to be used which sometimes is not possible. Also, Istio causes additional overhead like noise filtering
+- cumulativetodelta transformation not suppoerted
+  This transformation is mandatory for Dynatrace support and requires that data from a pod gets processed always by the same instance
+- General misconception
+  Gateways are usually running outside the cluster as a first entry point. The cluster should get rid of the data as fast as possible as persistence cannot be provided here (thinks like Kafka would again run outside). Try to compensate for short-lived hickups but otherwise just point back the backpressure to the client and do not try to solve the big picture
+
+## Proposal
+
+Most of the drawbacks can be solved by running the gateway logic node-local only. Every instance always processes only data of the local node. With that a natural scalability is given and can be extended by vertical scaling capabilities. The "old" gateways will be running in agent mode but still called "OTLP Gateway"
+
+![arch](./../assets/otlp-gateway-new.drawio.svg)
+
+The left drawbacks of the agent approach are:
+- coupling of signal types
+  The decoupling of the signal types will be lost, potentially leading to situations where the agent drops logs caused by metrics overload. However, with the right hardening that should not be a problem
+- Downtimes on updates
+  A rollout of the agent cannot happen in a rolling way, especially of the host port assignment. So there will be a downtime always which should be compoensated by the retry mechanism in the otel-sdk
+- host port conflicts
+  The user might serve already the OTLP ports on every port
+
+Additional benefits not outlined yet
+- persistent queue everywhere
+  A small persistent buffer could be used in all components based on the node file system
+- No Istio dependency
+  There is no Istio dependency at all anymore, no at the App and not at the components. However, the metric agent should still support scraping of istiofied endpoints
```

Comment:
```suggestion

  Neither the App nor the components have any Istio dependency anymore. However, the metric agent should still support the scraping of istiofied endpoints.
```
You did not mean " istiofied", did you? Should these be "istiofield" instead? I wasn't sure...

Reply:
The Istiofication of workloads is progressing fast, haven't you realized :) I meant it like that :)